framework/wazuh/mitre.py

```diff
@@ -1,141 +1,296 @@
-        for entry in self._data:
-            try:
-                # We find if we already have a dict with this id
-                found_in_pos = list(more_itertools.locate(result, pred=lambda d: d['id'] == entry['id']))[0]
-                for k, v in entry.items():
-                    if k not in unique_fields and v not in result[found_in_pos][k]:
-                        # We append the fields to their corresponding dict if id is already present
-                        result[found_in_pos][k].append(v)
-            except IndexError:
-                # We format and add a dict to result if id was not found
-                result.append(
-                    {k: (json.loads(v) if k == 'json' else v) if k in unique_fields else [v] for k, v in entry.items()})
-        self._data = result
-        return super()._format_data_into_dictionary()
+logger = logging.getLogger('wazuh')
 
 
 @expose_resources(actions=["mitre:read"], resources=["*:*:*"])
-def get_attack(id_: str = None, phase_name: str = None, platform_name: str = None, select: dict = None, search: dict = None,
-               offset: int = 0, limit: int = None, sort: dict = None, q: str = None, ) -> Dict:
-    """Get information from Mitre database.
-
-    :param id_: Filters by attack ID
-    :param phase_name: Filters by phase
-    :param platform_name: Filters by platform
-    :param search: Search if the string is contained in the db
-    :param offset: First item to return
-    :param limit: Maximum number of items to return
-    :param sort: Sort the items. Format: {'fields': ['field1', 'field2'], 'order': 'asc|desc'}
-    :param select: Select fields to return. Format: {"fields":["field1","field2"]}.
-    :param q: Query to filter by
-    :return: AffectedItemsWazuhResult with the data of the query from Mitre database
+def mitre_metadata() -> Dict:
+    """Return the metadata of the MITRE's database
+
+    Returns
+    -------
+    Metadata of MITRE's db
     """
-    result = AffectedItemsWazuhResult(all_msg='All selected MITRE information was returned',
-                                      none_msg='No MITRE information was returned'
-                                      )
-
-    # Set default limit to 500 if json is not selected and 10 otherwise in order to avoid congesting wdb socket
-    default_limit = 10 if select is None or 'json' in select else 500
-    limit = min(limit, default_limit) if limit is not None else default_limit
-
-    # Add regular field filters to q
-    if id_:
-        q = f'{q};id={id_}' if q else f'id={id_}'
-    if phase_name:
-        q = f'{q};phase_name={phase_name}' if q else f'phase_name={phase_name}'
-    if platform_name:
-        q = f'{q};platform_name={platform_name}' if q else f'platform_name={platform_name}'
-
-    # Execute query
-    db_query = WazuhDBQueryMitre(offset=offset, limit=limit, query=q, sort=sort, search=search, select=select)
+    result = AffectedItemsWazuhResult(none_msg='No metadata information was returned',
```

Comment:
Add MITRE to messages as specified in spec review

Reply:
done